src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -148,5 +177,235 @@ private void InitializeBuilder()
                     .WithObserverFactory(this);
             }
         }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        internal async Task<CosmosDBTriggerMetrics> GetMetricsAsync(IRemainingWorkEstimator workEstimator)
+        {
+            _workEstimator = workEstimator;
+            return await GetMetricsAsync();
+        }
+
+        public async Task<CosmosDBTriggerMetrics> GetMetricsAsync()
+        {
+            int partitionCount = 0;
+            long remainingWork = 0;
+            IReadOnlyList<RemainingPartitionWork> workPerPartition = null;
+
+            try
+            {
+                InitializeBuilder();
+
+                if (_workEstimator == null)
+                {
+                    _workEstimator = await this._hostBuilder.BuildEstimatorAsync();
+                }
+
+                workPerPartition = await this._workEstimator.GetEstimatedRemainingWorkPerPartitionAsync();
+
+                if (workPerPartition != null)
```

Comment:
When would this be null?

Reply:
You're right, it shouldn't be. I think this was an artifact of copying from ScaleController